Api testing

Eternal-Flame085 opened the pull request:
- Created spec for top movies
- Created route
- Created Movies controller and MoviesApiService controller
- Created view for top movies
- Needs more testing

Eternal-Flame085 committed:
created css

Eternal-Flame085 committed:
Created Movies controller

Eternal-Flame085 committed:
simple spec for movies controller

Eternal-Flame085 committed:
Created route for top rated

Eternal-Flame085 committed:
Created spec for top movies

Eternal-Flame085 committed:
View page for top rated movies

Eternal-Flame085 committed:
Changed route for top rated movies

Eternal-Flame085 committed:
Merger from main

Eternal-Flame085 committed:
Movie Api Service Class

Co-Authored-By: Sheryl Stillman <stillsheryl@users.noreply.github.com>

Eternal-Flame085 committed:
changed to figaro key
Co-authored-by: Sheryl Stillman 
<7945439+stillsheryl@users.noreply.github.com>

Eternal-Flame085 committed:
Renamed to better suited variable name


Co-authored-by: Sheryl Stillman 
<7945439+stillsheryl@users.noreply.github.com>

Co-Authored-By: Sheryl Stillman <stillsheryl@users.noreply.github.com>

Eternal-Flame085 committed:
Refactored the index and renamed @array to top movies

Co-authored-by: Sheryl Stillman 
<7945439+stillsheryl@users.noreply.github.com>

Co-Authored-By: Sheryl Stillman <stillsheryl@users.noreply.github.com>

Stearnzy commented on app/controllers/movies_controller.rb:
```diff
@@ -1,23 +1,9 @@
-      @titles << title_data[:results]
-    end
+    @titles = MovieApiService.movie_search(params[:movie_title])
+  end
```
Thanks for moving these

Stearnzy commented on spec/features/movie/movie_spec.rb.rb:
```diff
@@ -0,0 +1,27 @@
+require 'rails_helper'
```
Perhaps we can rename this spec file

stillsheryl commented on app/movie/movie_api_service.rb:
```diff
@@ -0,0 +1,46 @@
+        req.params[:query] = query
+      end
+
+      title_data = JSON.parse(response.body, symbolize_names: true)
```
Can we also use the parse_data method here?

Eternal-Flame085 committed:
Moved and renamed the movies_spec

Co-authored-by: Zach Stearns 
<64452036+Stearnzy@users.noreply.github.com>
Co-authored-by: Sheryl Stillman 
<7945439+stillsheryl@users.noreply.github.com>

Co-Authored-By: Zach Stearns <zach.stearns92@gmail.com>
Co-Authored-By: Sheryl Stillman <stillsheryl@users.noreply.github.com>

stillsheryl commented on spec/models/movie_api_spec.rb:
```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
```
I'm not sure what the convention is, but have seen something like this before:

```
describe MovieApiService do
  context "instance methods" do
    describe ".conn" do
      it "returns 200 status" do
        api_response = MovieApiService.conn.get("/3/movie/top_rated")
        expect(api_response.status).to eq(200)
      end
   end
end
```

stillsheryl reviewed (approved):
Looks good!
